### PR TITLE
Rename EntityExtractorMixin

### PR DESCRIPTION
**Pull request by VitorLamego**
Co-authore-by: AlGouvea <alvarohsgouvea@gmail.com>

**Proposed changes**:
- As proposed in the https://github.com/RasaHQ/rasa/issues/10225 references of EntityExtractorMixin were renamed to EntityExtractor for consistency.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by melindaloubser1 — CHANGES_REQUESTED** [review]
Thanks for addressing this!
The [migration guide in the docs](https://github.com/RasaHQ/rasa/blob/main/docs/docs/migration-guide.mdx#changes-to-custom-nlu-components) should also be updated to indicate the change. Since the migration will differ for users before and after the rename, I'd suggest adding a callout like e.g.
```
:::info Changed in 3.3

As of 3.3, the `EntityExtractorMixin` has been renamed back to `EntityExtractor`. The migration instructions below are valid for versions >= 3.0, <3.3.

:::

```

**Review by melindaloubser1 — APPROVED** [review]
🚀

File: changelog/10225.improvement.md
Comment (melindaloubser1) [inline changelog/10225.improvement.md:1]:
```suggestion
Rename `EntityExtractorMixin` to `EntityExtractor` for consistency, as was discussed in issue 10225.
```
a nitpick =)

**Comment by melindaloubser1** [comment]
Since I can't push to your fork, there's a branch & commit here with the suggested changes - could you pull these into your branch?
https://github.com/RasaHQ/rasa/commit/b1401df095049b04c230fbf0005a306ee39eb144

**Comment by VitorLamego** [comment]
Done!

**Review by melindaloubser1 — APPROVED** [review]

**Comment by melindaloubser1** [comment]
Thanks, do you mind addressing the[ failing CI checks](https://github.com/RasaHQ/rasa/runs/7680562006?check_suite_focus=true) ? They're my fault, I added the logging module when testing locally.
The documentation check is failing because of a flaky link, don't worry about that one.

**Comment by VitorLamego** [comment]
No problem! Is there any way I could help on documentation check ?? @melindaloubser1

**Comment by melindaloubser1** [comment]
There was a PR to make the check non-required here: https://github.com/RasaHQ/rasa/pull/11413 - the commit is also on `main`, so if you merge main into your branch, it should take care of it.

**Comment by melindaloubser1** [comment]
Last failing check here https://github.com/RasaHQ/rasa/runs/7748482684?check_suite_focus=true you can fix it by running `make formatter`

**Comment by melindaloubser1** [comment]
@VitorLamego I'm so sorry, now there's conflicts that have to be resolved - I'll try make sure this time once that's done this gets merged quickly so it doesn't get into a conflict state again

**Comment by VitorLamego** [comment]
Done @melindaloubser1 !

File: docs/docs/migration-guide.mdx
Comment (melindaloubser1) [inline docs/docs/migration-guide.mdx:43]:
```suggestion
```

Comment (melindaloubser1) [reply docs/docs/migration-guide.mdx:43]: This got in here by a matter of unfortunate timing - it's not in main anymore so it shouldn't be added

**Review by melindaloubser1 — CHANGES_REQUESTED** [review]

**Comment by VitorLamego** [comment]
Everything ok ? @melindaloubser1

**Review by ancalita — APPROVED** [review]
@VitorLamego Approving this as Melinda is no longer a maintainer for this repo.
Please run `make formatter` to resolve the formatting check fail. Before that ensure that the version of `black` in your venv is `22.6.0`.
Also you can run `make lint` and `make types` to check for any additional errors that this Code quality check could uncover and fix them before committing.

**Comment by VitorLamego** [comment]
Sorry for the delay! I believe it is ok now :) @ancalita

File: docs/docs/migration-guide.mdx
Comment (ancalita) [inline docs/docs/migration-guide.mdx:18]: Not entirely sure how this section ended up here, maybe you merged an old version of `main` into your branch, however this needs to be removed (Lines 18 - 43 inclusive).

File: rasa/nlu/extractors/extractor.py
Comment (ancalita) [inline rasa/nlu/extractors/extractor.py:481]: Code Quality check failed because the parameters in this method need to have type hints. Please add them, kwargs can be of type `Any`.

**Review by ancalita — CHANGES_REQUESTED** [review]
Looks good 👍🏻 
A few things need ironing out before merging, please address the comments below 🙏🏻

**Comment by VitorLamego** [comment]
It seems that errors on CI are not due to this issue, right ?? Anything else we could do to help ? @ancalita

**Comment by ancalita** [comment]
> It seems that errors on CI are not due to this issue, right ?? Anything else we could do to help ? @ancalita

I've restarted the jobs, could have been flaky incidents.

File: rasa/nlu/extractors/extractor.py
Comment (AlvaroLeles) [inline rasa/nlu/extractors/extractor.py:481]:
```suggestion
    def __init_subclass__(cls: EntityExtractor, **kwargs: Any):
```

**Review by AlvaroLeles — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
